osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+from datetime import date
+
+logger = logging.getLogger(__name__)
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+
+
+def delete_tasks_results(tasksToDelete):
+    from kobo.hub import models
+    for task in tasksToDelete:
+        try:
+            pathToDelete = models.Task.get_task_dir(task.id)
+            shutil.rmtree(pathToDelete)
+        except OSError as e:
+            logging.error("Error deleting tasks folder: %s", e)
+
+
+# In order to identify personal tasks, we are using the owner field. If the owner doesn't contain any / or @, we assume
+# that it is a personal task.
+def identify_personal_task(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough, and we are more perfomance-oriented.
+    return "/" not in owner and "@" not in owner
+
+
+def main():
+    import django
+    django.setup()
+
+    # Retention policy Task
+    logger.info("Starting Retention Policy Enforcement")
+
+    from osh.hub.scan import models
+    tasks = models.Task.objects.all()
+
+    # ---------- Section to take parameters from DB
+
+    retention_personal_tasks = models.RetentionPolicySetting.objects.get_days_from_category("PersonalScan")
+    retention_failed_status = models.RetentionPolicySetting.objects.get_days_from_category("FailedStatus")
+
+    # ---------- End Section to take parameters from DB
+
+    eligible_personal_tasks = []
+    eligible_failed_status = []
```

Comment:
As suggested by @kdudka in private, we can just use a single array to store the to-be-deleted tasks to simplify the code further.